tests/Dummy/ComplexObjectUncaster.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Spatie\DataTransferObject\Tests\Dummy;
+
+use InvalidArgumentException;
+use Spatie\DataTransferObject\Uncaster;
+
+class ComplexObjectUncaster implements Uncaster
+{
+    /**
+     * @param array|mixed $value
+     *
+     * @return mixed
+     */
+    public function uncast($value): array
+    {
+        if (! $value instanceof ComplexObject) {
```

Comment:
Can't provide a type hint since a child class can't be more specific in PHP. This is the best way to enforce it at run time and get typing.